app/helpers/encryption_helper.rb

```diff
@@ -0,0 +1,151 @@
+require "csv"
+module EncryptionHelper
+  def assign_secret_key
+    primes = []
+
+    CSV.foreach("primes_test.csv") do |row|
+      primes << row.first.to_i
+    end
+
+    length = primes.length - 1
+
+    key = [primes[rand(length)], primes[rand(length)]]
+
+    while key[0] == key[1]
+      key = [primes[rand(length)], primes[rand(length)]]
+    end
+
+    key
+  end
+
+  def euclidean_algorithm(a, b)
+    euclidean_algorithm = [[a, a/b, b, a%b]]
+    n = 0
+    while euclidean_algorithm[n][3] > 0
+      iteration = [euclidean_algorithm[n][2],
+                   euclidean_algorithm[n][2] / euclidean_algorithm[n][3],
+                   euclidean_algorithm[n][3],
+                   euclidean_algorithm[n][2] % euclidean_algorithm[n][3]]
+      euclidean_algorithm << iteration
+
+      n += 1
+    end
+    euclidean_algorithm
+  end
+
+  def gcd(a, b)
+    e_a = euclidean_algorithm(a,b)
+    if e_a.length > 1
+      e_a[e_a.length - 2][3]
+    else
+      e_a[0][2]
+    end
+  end
+
+  def calculate_public_key(p, q)
+    m = p * q
+    phi_of_m = (p-1)*(q-1)
+
+    k = rand(10000000)
+
+    while gcd(phi_of_m, k) != 1
+      k = rand(10000000)
+    end
+
+    [m,k]
+  end
+
+  def encoder_hash
+    { "a" => "11", "b" => '12', "c" => "13", "d" => "14", "e" => "15",
+      "f" => "16", "g" => '17', "h" => "18", "i" => "19", "j" => "20",
+      "k" => "21", "l" => '22', "m" => "23", "n" => "24", "o" => "25",
+      "p" => "26", "q" => '27', "r" => "28", "s" => "29", "t" => "30",
+      "u" => "31", "v" => '32', "w" => "33", "x" => "34", "y" => "35",
+      "z" => "36"
+    }
+  end
+
+  def decoder_hash
+    decoder_hash = {}
+
+    encoder_hash.each do |k, v|
+      decoder_hash[v] = k
+    end
+    decoder_hash
+  end
+
+  def encode(message)
+
+    #strips message and puts each letter into it's own space in an array
+    message_array = message.downcase.gsub(/[^a-z]/, "").split("")
+    encoded_message = []
+    message_array.each do |letter|
+      encoded_message << encoder_hash[letter]
+    end
+    encoded_message.join('')
+    #returns encoded message as a string of digits
+
+  end
+
+  def split_message(message, m)
+    x = 0
+    split_message = []
+    while x < encode(message).length
+      split_message << encode(message).slice(x, m-1)
+      x += (m-1)
+    end
+  end
+
+  def successive_squaring(number, m, k)
+    x_array = []
+    while k > 0
+      x = Math.log2(k).to_i
+      x_array << x
+      k -= 2**x
+    end
+
+    successive_squaring = [number]
+    x = x_array.first
+    x.times do
+      number = number**2 % m
+      successive_squaring << number
+    end
+
+    product = 1
+
+    x_array.each do |y|
```

Comment:
Unused block argument - `y`. You can omit the argument if you don't care about it.